src/compas_timber/connections/l_halflap.py

```diff
@@ -31,25 +33,71 @@ class LHalfLapJoint(LapJoint):
+    flip_lap_side : bool
+        If True, the lap is flipped to the other side of the beams.
+    cut_plane_bias : float
+        Allows lap to be shifted deeper into one beam or the other. Value should be between 0 and 1.0 without completely cutting through either beam. Default is 0.5.
     """
 
     SUPPORTED_TOPOLOGY = JointTopology.TOPO_L
 
-    def __init__(self, main_beam=None, cross_beam=None, flip_lap_side=False, cut_plane_bias=0.5, **kwargs):
-        super(LHalfLapJoint, self).__init__(main_beam, cross_beam, flip_lap_side, cut_plane_bias, **kwargs)
+    @property
+    def __data__(self):
+        data = super(LHalfLapJoint, self).__data__
+        data["beam_a"] = self.beam_a_guid
+        data["beam_b"] = self.beam_b_guid
+        data["flip_lap_side"] = self.flip_lap_side
+        data["cut_plane_bias"] = self.cut_plane_bias
+        return data
+
+    def __init__(self, beam_a=None, beam_b=None, flip_lap_side=None, cut_plane_bias=None, **kwargs):
+        super(LHalfLapJoint, self).__init__(**kwargs)
+        self.beam_a = beam_a
+        self.beam_b = beam_b
+        self.beam_a_guid = kwargs.get("beam_a_guid", None) or str(beam_a.guid)
+        self.beam_b_guid = kwargs.get("beam_b_guid", None) or str(beam_b.guid)
+
+        self.flip_lap_side = flip_lap_side
+        self.cut_plane_bias = 0.5 if cut_plane_bias is None else cut_plane_bias
+        self.features = []
+
+    @property
+    def elements(self):
+        return [self.beam_a, self.beam_b]
+
+    @property
+    def beam_a_ref_side_index(self):
+        cross_vector = self.beam_a.centerline.direction.cross(self.beam_b.centerline.direction)
+        ref_side_dict = beam_ref_side_incidence_with_vector(self.beam_a, cross_vector, ignore_ends=True)
+        if self.flip_lap_side:
+            return max(ref_side_dict, key=ref_side_dict.get)
+        return min(ref_side_dict, key=ref_side_dict.get)
+
+    @property
+    def beam_b_ref_side_index(self):
+        cross_vector = self.beam_a.centerline.direction.cross(self.beam_b.centerline.direction)
+        ref_side_dict = beam_ref_side_incidence_with_vector(self.beam_b, cross_vector, ignore_ends=True)
+        if self.flip_lap_side:
+            return min(ref_side_dict, key=ref_side_dict.get)
+        return max(ref_side_dict, key=ref_side_dict.get)
+
+    @property
+    def cutting_plane_a(self):
+        # the plane that cuts beam_a as a planar surface
+        ref_side_dict = beam_ref_side_incidence(self.beam_a, self.beam_b, ignore_ends=True)
+        ref_side_index = max(ref_side_dict, key=ref_side_dict.get)
+        return self.beam_b.side_as_surface(ref_side_index)
+
+    @property
+    def cutting_plane_b(self):
+        # the plane that cuts beam_b as a planar surface
+        ref_side_dict = beam_ref_side_incidence(self.beam_b, self.beam_a, ignore_ends=True)
+        ref_side_index = max(ref_side_dict, key=ref_side_dict.get)
+        return self.beam_a.side_as_surface(ref_side_index)
```

Comment:
these look very similar to the ones above, are they the same.
if we don't want these in a parent class, maybe they could be moved into a new higher level function(s) in our joinery toolkit modules where they return what's needed (index of surface) instead of the dictionary.

Then they can be called once in `add_features` this will be more readable, will have less duplication code and will potentially be more performant as some of these properties end up being called multiple times.